Userland/Demos/CatDog/CatDog.cpp

```diff
@@ -10,174 +10,188 @@
-        }
-    }
+    if (maybe_proc_info.is_error())
+        return {};
+
+    auto proc_info = maybe_proc_info.release_value();
+    uint16_t special_state {};
+
+    auto maybe_paint_program = proc_info.processes.first_matching([](auto& process) {
+        return process.name.equals_ignoring_case("pixelpaint"sv) || process.name.equals_ignoring_case("fonteditor"sv);
+    });
+    if (maybe_paint_program.has_value())
+        special_state |= State::ARTIST;
 
+    auto maybe_inspector_program = proc_info.processes.first_matching([](auto& process) {
+        return process.name.equals_ignoring_case("inspector"sv) || process.name.equals_ignoring_case("systemmonitor"sv) || process.name.equals_ignoring_case("profiler"sv);
+    });
+    if (maybe_inspector_program.has_value())
+        special_state |= State::INSPECTOR;
+
+    return special_state;
+}
+
+bool CatDog::is_artist() const
+{
+    return (special_application_states() & State::ARTIST) != 0;
+}
+
+bool CatDog::is_inspector() const
+{
+    return (special_application_states() & State::INSPECTOR) != 0;
+}
+
+void CatDog::timer_event(Core::TimerEvent&)
+{
+    m_state = State::IDLE | special_application_states();
     if (!m_roaming)
         return;
-    if (m_temp_pos.x() > 48) {
-        m_left = false;
-        m_right = true;
-        m_moveX = 16;
-
-        m_curr_bmp = m_erun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_erun2;
-    } else if (m_temp_pos.x() < -16) {
-        m_left = true;
-        m_right = false;
-        m_moveX = -16;
-
-        m_curr_bmp = m_wrun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_wrun2;
-    } else {
-        m_left = false;
-        m_right = false;
-        m_moveX = 0;
-    }
 
-    if (m_temp_pos.y() > 48) {
-        m_up = false;
-        m_down = true;
-        m_moveY = 10;
-
-        m_curr_bmp = m_srun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_srun2;
-    } else if (m_temp_pos.y() < -16) {
-        m_up = true;
-        m_down = false;
-        m_moveY = -10;
-
-        m_curr_bmp = m_nrun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_nrun2;
-    } else {
-        m_up = false;
-        m_down = false;
-        m_moveY = 0;
-    }
+    Gfx::IntPoint move;
+    auto const size = window()->size();
 
-    if (m_up && m_left) {
-        m_curr_bmp = m_nwrun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_nwrun2;
-    } else if (m_up && m_right) {
-        m_curr_bmp = m_nerun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_nerun2;
-    } else if (m_down && m_left) {
-        m_curr_bmp = m_swrun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_swrun2;
-    } else if (m_down && m_right) {
-        m_curr_bmp = m_serun1;
-        if (m_curr_frame == 2)
-            m_curr_bmp = m_serun2;
+    if (m_mouse_offset.x() < 0) {
+        m_state |= State::LEFT;
+        move.set_x(max(m_mouse_offset.x(), -size.width() / 2));
+
```

Comment:
Are these empty lines intentional? We generally don't have empty lines particularly at the end of a block

Reply:
Yes, they were intentional. Force of habit. At Google, "else if" blocks are often highlighted with an empty line. Especially if the blocks are basically independent from the "if" condition (as in, the "else" is not really needed).
But I've removed the lines to match Serenity standards. Done.